src/main/scala/net/ripe/rpki/publicationserver/StateActor.scala

```diff
@@ -104,16 +110,20 @@ class StateActor(conf: AppConfig) extends Actor with Hashing with Logging {
     }
   }
 
-  private def applyReplace(state: State, clientId: ClientId, uri: URI, hashToReplace: String, bytes: Bytes): Either[ReportError, State] = {
+  private def applyReplace(state: State, 
+                           clientId: ClientId, 
+                           uri: URI, 
+                           hashToReplace: String, 
+                           newBytes: Bytes): Either[ReportError, State] = {
     state.get(uri) match {
       case Some((_, Hash(foundHash), _)) =>
-        if (foundHash.toUpperCase == hashToReplace.toUpperCase)
-          Right(state + (uri -> (bytes, hash(bytes), clientId)))
-        else
         if (foundHash.toUpperCase == hashToReplace.toUpperCase) {
-          val newHash = hash(bytes)
+          val newHash = hash(newBytes)
           logger.debug(s"Replacing $uri with hash $foundHash -> $newHash")
-          Right(state + (uri -> (bytes, newHash, clientId)))
+          // one added one deleted
+          metrics.publishedObject()
+          metrics.withdrawnObject()
+          Right(state + (uri -> (newBytes, newHash, clientId)))
         } else
           Left(ReportError(BaseError.NonMatchingHash,
             Some(s"Cannot republish the object [$uri], hash doesn't match, passed ${hashToReplace.toUpperCase}, but existing one is ${foundHash.toUpperCase}.")))
```

Comment:
Do we want to count these failures?

Reply:
Yes, I think so. Will add counter for then as well.